app/controllers/users_controller.rb

```diff
@@ -496,28 +498,61 @@ def is_local_username
     usernames -= groups
     usernames.each(&:downcase!)
 
-    cannot_see = []
+    users = User
+      .where(staged: false)
+      .where(username_lower: usernames)
+      .index_by(&:username_lower)
```

Comment:
```suggestion
    users = User
      .where(staged: false, username_lower: usernames)
      .index_by(&:username_lower)
```

Minor but we can avoid an extra method call here.